src/Avalonia.Controls/StatusBar.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System.Collections;
+using Avalonia.Controls.Generators;
+using Avalonia.Controls.Presenters;
+using Avalonia.Controls.Primitives;
+using Avalonia.Controls.Templates;
+using Avalonia.Input;
+
+namespace Avalonia.Controls
+{
+    /// <summary>
+    /// An <see cref="ItemsControl"/> in which individual items can be selected.
+    /// </summary>
+    public class StatusBar : ItemsControl
```

Comment:
I think you also need to override [CreateItemContainerGenerator like this](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/ListBox.cs#L88) to make sure that when the `StatusBar` is data-bound it will generate `StatusBarItem`s.

Reply:
thanks .. i totally missed this ...  i created overrid CreateItemContainerGenerator and created relevant tests